tests/system/Validation/ValidationTest.php

```diff
@@ -628,6 +628,33 @@ public function testTagReplacement()
         $this->assertSame($expected, $errors['Username']);
     }
 
+    public function testRulesForObjectField()
+    {
+        $config          = new App();
+        $config->baseURL = 'http://example.com/';
+
+        $this->validation->setRules([
+            'configuration' => 'required|check_object_rule',
+        ]);
+
+        $data = (object) ['configuration' => (object) ['first' => 1, 'second' => 2]];
+        $this->validation->run((array) $data);
```

Comment:
Why do you need to cast this first to object then recast to array? Casting is not recursive. https://3v4l.org/gMad5
```suggestion
        $data = ['configuration' => (object) ['first' => 1, 'second' => 2]];
        $this->validation->run($data);
```

Reply:
If you look at the method, it only accepts arrays.
I Know that isn't recursive. Configuration attribute is a Object.

Reply:
The change that I propose can be used to validate that what an object has are the mandatory fields, for example:

_**public function check_token_configuration( Object $value, string $socialNetwork, array $data = [] )
    {
        $doctrine = \Config\Services::doctrine();

        if( !isset( $data[ $socialNetwork ] ) ) { return false; }

        $configSocialConnection = config( \ucfirst( $data[ $socialNetwork ] ) . '\\Connection' );
        
        foreach( \get_object_vars( $configSocialConnection ) as $key => $val )
        {
            if( !isset( $value->{ $key } ) )
            {
                return false;
            }

            if( empty( $value->{ $key } ) ){ return false; }

            $configSocialConnection->{ $key } = $value->{ $key };
        }

        return true;
    }**_


Without the change that I propose it does not work

Reply:
Hmm. Fair enough. Thanks for the explanation.